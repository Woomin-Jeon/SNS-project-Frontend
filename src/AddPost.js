/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable no-alert */
/* eslint-disable no-shadow */
import React, { useState } from 'react';
import { addPost } from './function';
import ShowPost from './ShowPost';

const initialPost = {
  temptId: '',
  temptContents: '',
  post: [
    {
      id: '홍길동', // 작성자
      contents: '오늘은 날씨가 쌀쌀하니 좋아요 기능을 넣어보면 좋을 것 같다', // 게시글의 내용
      thumbCount: 0, // 좋아요 갯수
      sharingCount: 0, // 공유 횟수
      commentCount: 0, // 댓글 갯수
      comment: '',
    },
  ],
};
/* 댓글 입력 버튼을 누르면, 해당 게시글의 id값을 onclick함수의 매개변수로 넣어주고 id를 포함한 객체를 comment
배열에 넣어준다. 그리고 post.map에서 post.id == comment.id 인 것을 찾아서 이것의 statement를 렌더링 해준다.
*/
function AddPost() {
  const [poststate, setPostState] = useState(initialPost);
  const { temptId, temptContents } = poststate;

  const setPostTemptId = (temptId) => {
    setPostState({ ...poststate, temptId });
  };
  const setPostTemptContents = (temptContents) => {
    setPostState({ ...poststate, temptContents });
  };

  const handleAddPost = () => {
    if (temptId.trim() && temptContents.trim()) {
      setPostState({ ...addPost(poststate, temptId, temptContents), temptId: '', temptContents: '' });
    } else {
      alert('내용을 입력해주세요');
    }
  };

  return (
    <>
      <div className="addpost">
        <div className="addpost-title">게시물 만들기</div>
        <div>작성자: <input className="addpost-writer" type="text" value={temptId} onChange={(e) => setPostTemptId(e.target.value)} /></div>
        <div>내용: <input className="addpost-inputcontents" type="text" value={temptContents} onChange={(e) => setPostTemptContents(e.target.value)} /></div>
        <button className="addpost-out" type="button" onClick={handleAddPost}>게시</button>
      </div>
      <ShowPost poststate={poststate} />
    </>
  );
}

export default AddPost;
