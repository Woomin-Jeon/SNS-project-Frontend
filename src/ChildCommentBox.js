/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import { addChildComment } from './function';

function ChildCommentBox({
  commentState,
  setCommentState,
  parentsComment,
  currentUserState,
}) {
  const { id, userName } = currentUserState;
  const [temptState, setTemptState] = useState('');

  const handleAddChildComment = () => {
    if (temptState.trim()) {
      setCommentState(addChildComment(commentState, temptState, id, userName, parentsComment));
      setTemptState('');
    }
  };

  return (
    <div className="child-comment">
      └ {userName} : <input type="text" value={temptState} onChange={(e) => setTemptState(e.target.value)} />
      <button type="button" className="comment-input" onClick={handleAddChildComment}>입력</button>
    </div>
  );
}

export default ChildCommentBox;
