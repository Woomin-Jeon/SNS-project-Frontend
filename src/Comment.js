/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/jsx-wrap-multilines */
/* eslint-disable implicit-arrow-linebreak */
import React from 'react';
import profile from './profile.jpeg';

const Comment = ({ commentState, post }) => {
  const { comment } = commentState;
  const appropriateComment = [];

  for (let i = 0; i < comment.length; i++) {
    if (post.id === comment[i].id) {
      appropriateComment.push(comment[i]);
    }
  }

  return (
    <div className="comment-contents">
      {appropriateComment.map((v) =>
        <div key={v.writer}>
          <span className="comment-main">
            <img className="comment-image" src={profile} alt="" width="3.5%" />
            {v.writer} : {v.statement}
          </span>
          <button type="button" className="comment-thumb">좋아요</button>
          <span className="comment-thumb-count">{v.commentsThumbCount}</span>
        </div>)}
    </div>
  );
};

export default Comment;
