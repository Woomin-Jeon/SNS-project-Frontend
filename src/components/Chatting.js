import React, { useState, useEffect } from 'react';
import { findUserById } from '../function';

function Chatting({
  setIsChattingOn,
  currentUserState,
  socket,
  userSocketID,
  setUserSocketID,
  loginState,
  friendID,
  setFriendID,
}) {
  const [message, setMessage] = useState('');
  const [chattingMessages, setChattingMessages] = useState([]);
  const { users } = loginState;

  useEffect(() => {
    socket.on('hello', (data) => {
      console.log('Message : ', data.message);
      setChattingMessages((chattingMessages) =>
        [...chattingMessages, { id: data.userID, msg: data.message }]);
    });
  }, []);

  const getUserInputMessage = (contents) => {
    setMessage(contents);
  };

  const sendMessage = () => {
    const userID = currentUserState.id;

    socket.emit('chat message', { message, userSocketID, userID });
    setChattingMessages([...chattingMessages, { id: currentUserState.id, msg: message }]);
    setMessage('');
  };

  const closeChatting = () => {
    setUserSocketID('');
    setFriendID('');
    setIsChattingOn(false);
  };

  return (
    <div className="chatting">
      <div className="chatting-header">
        {findUserById(users, friendID).userName}
        님 과의 채팅방입니다
      </div>
      <button className="chatting-exit" type="button" onClick={closeChatting}>X</button>
      <div>
        {chattingMessages.map((v, i) => (
          <div key={i}>
            <div className="chatting-message-body">
              <span className="chatting-partner-name">{findUserById(users, v.id).userName}</span>
              <span className="chatting-message">{v.msg}</span>
            </div>
          </div>
        ))}
      </div>
      <div className="chatting-input">
        <input type="text" value={message} onChange={(e) => getUserInputMessage(e.target.value)} />
        <button type="button" className="chatting-send-button" onClick={sendMessage}>전송</button>
      </div>
    </div>
  );
}

export default Chatting;
