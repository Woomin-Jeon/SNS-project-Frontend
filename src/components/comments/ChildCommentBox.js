import React, { useState } from 'react';
import { addChildComment } from '../../function';

function ChildCommentBox({
  commentState,
  setCommentState,
  parentsComment,
  currentUserState,
}) {
  const { id, userName } = currentUserState;
  const { uniqueKey } = parentsComment;
  const [temptState, setTemptState] = useState('');

  const handleAddChildComment = async () => {
    if (temptState.trim()) {
      const { postComments } = await addChildComment(uniqueKey , temptState, id, userName);

      setCommentState({ ...commentState, comment : [...postComments.comment]});
    }
  };

  return (
    <div className="child-comment">
      └ {userName} : <input type="text" value={temptState} onChange={(e) => setTemptState(e.target.value)} />
      <button type="button" className="comment-input" onClick={handleAddChildComment}>입력</button>
    </div>
  );
}

export default ChildCommentBox;
