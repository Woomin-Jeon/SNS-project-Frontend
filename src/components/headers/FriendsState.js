import React from 'react';
import { addFriend, removeFriend } from '../../function';

function FriendsState({
  specificPost,
  currentUserState,
  setCurrentUserState,
}) {
  const { id, name } = specificPost;
  const { friends } = currentUserState;

  const friendAddButtonClicked = () => {
    setCurrentUserState(addFriend(currentUserState, id));
  };
  const friendRemoveButtonClicked = () => {
    setCurrentUserState(removeFriend(currentUserState, id));
  };

  return (
    <>
      <h3>{name} 님의 타임라인</h3>
      <div>
        {friends.includes(id)
          ? (
            <div className="other-friend-contents">
              <span>{name} 님과 친구입니다</span>{' '}
              <button
                className="others-friend-already"
                type="button"
                onClick={friendRemoveButtonClicked}
              >
              친구해제
              </button>
            </div>
          ) : (
            <button
              className="others-friend-add"
              type="button"
              onClick={friendAddButtonClicked}
            >
            친구추가
            </button>
          )}
      </div>
    </>
  );
}

export default FriendsState;
