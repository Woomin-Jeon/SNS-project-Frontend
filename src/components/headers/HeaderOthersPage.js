/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable object-curly-newline */
/* eslint-disable react/prop-types */
/* eslint-disable no-alert */
import React, { useState } from 'react';
import { Redirect } from 'react-router-dom';

function HomeHeader({
  loginState,
  setLoginState,
  currentUserState,
  setCurrentUserState,
}) {
  const [myPageState, setMyPageState] = useState(false);
  const [timelinePageState, setTimelinePageState] = useState(false);
  const { isLoggedIn } = loginState;
  const { userName } = currentUserState;

  const moveToTimelinePage = () => {
    setTimelinePageState(true);
  };

  const moveToMyPage = () => {
    setMyPageState(true);
  };

  const logoutButtonClicked = () => {
    setLoginState({ ...loginState, isLoggedIn: false, temptId: '', temptPw: '' });
    setCurrentUserState({ ...currentUserState, id: '', pw: '', userName: '' });
    alert('로그아웃 되었습니다');
  };

  if (timelinePageState === true) {
    return <Redirect to="/post" />;
  }
  if (myPageState === true) {
    return <Redirect to="/mypage" />;
  }
  if (isLoggedIn === false) {
    return <Redirect to="/" />;
  }

  return (
    <>
      <h1>Facebook</h1>
      <span>{userName}{' '}</span>
      <button className="page-header" type="button" onClick={moveToTimelinePage}>홈</button>{' '}
      <button className="page-header" type="button" onClick={moveToMyPage}>마이페이지</button>{' '}
      <button className="page-header" type="button" onClick={logoutButtonClicked}>로그아웃</button>
      <br />
    </>
  );
}

export default HomeHeader;
