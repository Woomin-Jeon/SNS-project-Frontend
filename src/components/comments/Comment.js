import React, { useState, useEffect } from 'react';
import func from '../../function';
import ChildCommentBox from './ChildCommentBox';

const callAPI = async (commentState, setCommentState, loginState, setLoginState) => {
  const { postComments } = await func.getComments();
  const { userStore } = await func.getUsers();

  setLoginState({ ...loginState, users: [...userStore] });

  setCommentState({
    ...commentState,
    comment: [...postComments.reverse()],
  });
}

function Comment({
  specificPost,
  postState,
  setPostState,
  currentUserState,
  commentState,
  setCommentState,
  loginState,
  setLoginState,
}) {
  const { comment } = commentState;
  const { userName, id } = currentUserState;
  const { uniqueKey } = specificPost;
  const { users } = loginState;
  const [temptState, setTemptState] = useState('');

  useEffect(() => {
    callAPI(commentState, setCommentState, loginState, setLoginState);
  }, []);

  const setCommentTemptStatement = (temptState) => {
    setTemptState(temptState);
  };

  const handleAddComment = async () => {
    if (temptState.trim()) {
      const { postComments } = await func.addComment(uniqueKey, id, userName, temptState);
      const { timeLinePosts } = await func.plusCommentCount(uniqueKey);

      setCommentState({ ...commentState, comment: [...postComments.reverse()] });
      setPostState({ ...postState, post: [...timeLinePosts.reverse()] });
      setTemptState('');
    }
  };

  const handleCommentThumbCount = async (specificComment) => {
    const { postComments } = await func.plusCommentThumbCount(
      specificComment.uniqueKey,
      currentUserState.id,
    );

    setCommentState({ ...commentState, comment: [...postComments.reverse()] });
  };

  const ChildCommentButtonClicked = (specificComment) => {
    setCommentState(func.openChildCommentBox(commentState, specificComment));
  };
  return (
    <>
      <div className="comment">
        <span className="comment-word">댓글달기</span>
        <input
          className="comment-input-box"
          type="text"
          value={temptState}
          onChange={(e) => setCommentTemptStatement(e.target.value)}
        />
        <button
          className="comment-button"
          type="button"
          onClick={() => handleAddComment()}
        >
          입력
        </button>
      </div>
      <div className="comment-contents">
        {comment.filter((v) => specificPost.uniqueKey === v.id).map((v, index) =>
          <div className="comment-each" key={`Comment${index}`}>
            <span>
              <img
                className="comment-profile-image"
                src={users[users.findIndex(user => user.id === v.writerID)].profile}
                alt=""
                width="3.5%"
              />
              <div className="comment-statement">
                <span className="comment-writer">{v.writer}</span>
                <span className="comment-colon">:</span>
                <span className="comment-saying">{v.statement}</span>
              </div>
            </span>
            <button
              className="comment-thumb"
              type="button"
              onClick={() => ChildCommentButtonClicked(v)}
            >
              대댓글
            </button>
            <button
              type="button"
              className="comment-thumb"
              onClick={() => handleCommentThumbCount(v)}
            >
              좋아요
            </button>
            <span className="comment-thumb-count">{v.commentThumbCount.length}</span>
            {v.isChildCommentFunctionOn
              ? (
                <div className="child-comment-main">
                  <ChildCommentBox
                    commentState={commentState}
                    setCommentState={setCommentState}
                    parentsComment={v}
                    currentUserState={currentUserState}
                  />
                </div>
              ) : <></>}
            <div>
              {v.childComment.map((k, index) =>
                <div key={index} className="child-contents">
                  └ {k.name} :<span> </span>{k.statement}
                </div>)}
            </div>
          </div>)}
      </div>
    </>
  );
}

export default Comment;
