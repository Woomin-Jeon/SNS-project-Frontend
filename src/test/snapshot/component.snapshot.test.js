import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import AddPost from '../../components/posts/AddPost';
import PostEditBox from "../../components/posts/PostEditBox";
import ShowPostOthersPage from "../../components/posts/ShowPost";
import ShowPostHome from "../../components/posts/ShowPostHome";
import FriendsState from "../../components/headers/FriendsState";
import HomeHeader from "../../components/headers/HeaderHome";
import MyPageHeader from "../../components/headers/HeaderMyPage";
import OtherspageHeader from "../../components/headers/HeaderOthersPage";
import HeaderOthersPage from "../../components/headers/HeaderOthersPage";
import SearchBox from "../../components/headers/SearchBox";
import Comment from "../../components/comments/Comment";
import ChildCommentBox from "../../components/comments/ChildCommentBox";
import Chatting from "../../components/Chatting";

describe('Snapshot test', () => {
  let currentUserState;
  let setCurrentUserState;
  let postState;
  let setPostState;
  let specificPost;
  let commentState;
  let setCommentState;
  let loginState;
  let setLoginState;
  let setTopLevelState;
  let setSearchState;
  let setIsChattingOn;
  let userSocketID;
  let setUserSocketID;
  let friendID;
  let setFriendID;
  let socket;
  let p;
  let index;

  beforeEach(() => {
    specificPost = {
      id: 'TEST',
      name: 'TEST'
    };
    currentUserState = {
      id: 'TEST',
      pw: 'TEST',
      userName: 'TEST',
      friends: []
    };
    loginState = {
      users: [ { id: 'TEST' }]
    };
    commentState = {
      comment: []
    };
    index = 100;
    p = {
      time: [2020, 4, 21, 5, 16],
      thumbCount: [],
    };
    friendID = 'TEST';
  });


  describe('Chatting', () => {
    it('renders well', () => {
      const component = shallow(
        <Chatting
          setIsChattingOn={setIsChattingOn}
          currentUserState={currentUserState}
          socket={socket}
          userSocketID={userSocketID}
          setUserSocketID={setUserSocketID}
          loginState={loginState}
          friendID={friendID}
          setFriendID={setFriendID}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('ChildCommentBox', () => {
    it('renders well', () => {
      const component = shallow(
        <ChildCommentBox
          commentState={commentState}
          setCommentState={setCommentState}
          parentsComment={p}
          currentUserState={currentUserState}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('Comment', () => {
    it('renders well', () => {
      const component = shallow(
        <Comment
          specificPost={p}
          setPostState={setPostState}
          postState={postState}
          currentUserState={currentUserState}
          commentState={commentState}
          setCommentState={setCommentState}
          loginState={loginState}
          setLoginState={setLoginState}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('SearchBox', () => {
   it('renders well', () => {
     const component = shallow(
       <BrowserRouter>
         <SearchBox
           loginState={loginState}
           setLoginState={setLoginState}
           setSearchState={setSearchState}
         />
       </BrowserRouter>
     );

     expect(component).toMatchSnapshot();
   });
  });

  describe('HeaderOthersPage', () => {
    it('renders well', () => {
      const component = shallow(
        <BrowserRouter>
          <HeaderOthersPage
            loginState={loginState}
            setLoginState={setLoginState}
            currentUserState={currentUserState}
            setCurrentUserState={setCurrentUserState}
          />
        </BrowserRouter>
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('OtherspageHeader', () => {
    it('renders well', () => {
      const component = shallow(
        <BrowserRouter>
          <OtherspageHeader
            loginState={loginState}
            setLoginState={setLoginState}
            currentUserState={currentUserState}
            setCurrentUserState={setCurrentUserState}
          />
        </BrowserRouter>
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('MyPageHeader', () => {
    it('renders well', () => {
      const component = shallow(
        <BrowserRouter>
          <MyPageHeader
            loginState={loginState}
            setLoginState={setLoginState}
            currentUserState={currentUserState}
            setCurrentUserState={setCurrentUserState}
          />
        </BrowserRouter>
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('HomeHeader', () => {
    it('renders well', () => {
      const components = shallow(
        <BrowserRouter>
          <HomeHeader
            loginState={loginState}
            setLoginState={setLoginState}
            currentUserState={currentUserState}
            setCurrentUserState={setCurrentUserState}
          />
        </BrowserRouter>
      )

      expect(components).toMatchSnapshot();
    })
  })

  describe('FriendsState', () => {
    it('renders well', () => {
      const component = shallow(
        <FriendsState
          specificPost={specificPost}
          currentUserState={currentUserState}
          setCurrentUserState={setCurrentUserState}
          loginState={loginState}
          setLoginState={setLoginState}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('AddPost', () => {
    it('renders well', () => {
      const component = shallow(
        <AddPost
          currentUserState={currentUserState}
          postState={postState}
          setPostState={setPostState}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('PostEditBox', () => {
    it('renders well', () => {
      const component = shallow(
        <PostEditBox
          specificPost={specificPost}
          postState={postState}
          setPostState={setPostState}
          currentUserState={currentUserState}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('ShowPostOthersPage', () => {
    it('renders well', () => {
      const component = shallow(
        <ShowPostOthersPage
          postState={postState}
          setPostState={setPostState}
          currentUserState={currentUserState}
          commentState={commentState}
          setCommentState={setCommentState}
          p={p}
          index={index}
          loginState={loginState}
          setLoginState={setLoginState}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('ShowPostHome', () => {
    it('renders well', () => {
      const component = shallow(
        <BrowserRouter>
          <ShowPostHome
            postState={postState}
            setPostState={setPostState}
            currentUserState={currentUserState}
            commentState={commentState}
            setCommentState={setCommentState}
            setTopLevelState={setTopLevelState}
            specificPost={p}
            index={index}
            loginState={loginState}
            setLoginState={setLoginState}
          />
        </BrowserRouter>
      );

      expect(component).toMatchSnapshot();
    });
  });

});
