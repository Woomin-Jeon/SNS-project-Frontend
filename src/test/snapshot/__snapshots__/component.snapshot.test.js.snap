// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test AddPost renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test Chatting renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test ChildCommentBox renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test Comment renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test FriendsState renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test HeaderOthersPage renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test HomeHeader renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test MyPageHeader renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test OtherspageHeader renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test PostEditBox renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test SearchBox renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test ShowPostHome renders well 1`] = `ShallowWrapper {}`;

exports[`Snapshot test ShowPostOthersPage renders well 1`] = `ShallowWrapper {}`;
